### PR TITLE
Add RHE Linux 8 stream and AWS linux to software requirement doc

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
Add RHE Linux 8 stream and AWS linux to software requirement doc

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/11459
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by dveeden — CHANGES_REQUESTED** [review]
This isn't correct:

- "CentOS 8 Stream" doesn't have minor versions like 8.1, 8.2, etc. as it is a rolling release
- "Red Hat Enterprise Linux" doesn't have a stream version. RHEL 8 has minor versions like 8.0, 8.1, etc.
- As both CentOS 7 and RHEL 7 and other related distributions (OEL7) were all using minor versions it was easy to group/compare them. That's no longer true with CentOS Stream and RHEL.
- "AWS Linux" doesn't exist. "Amazon Linux" does. 
- For "Amazon Linux" we should only support version 2 as there are huge differences between Amazon Linux 2 and Amazon Linux 2022. While Amazon Linux 2 is compatible with RHEL7/CentOS7, Amazon Linux 2022 is built with Fedora Linux as a base, but with many big enhancements on top of that.

References:

- https://aws.amazon.com/linux/amazon-linux-2022/
- https://aws.amazon.com/amazon-linux-2/
- https://docs.aws.amazon.com/linux/al2022/ug/compare-al2-to-AL2022.html

And to try and clarify the RHEL/CentOS relationship (excluding CentOS Linux 8, which is EOL):

CentOS_RHEL_relateion

File: hardware-and-software-requirements.md [hunk 1]
```diff
@@ -19,7 +19,9 @@ Starting from v6.1.1, TiDB provides multi-level support for different quality st
     |  Operating systems   |   Supported CPU architectures   |
     |   :---   |   :---   |
     | <ul><li>Red Hat Enterprise Linux 8.4 or a later 8.x version</li><li>CentOS 8.4 or a later 8.x version</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
+    | Red Hat Enterprise Linux 8 Stream | <ul><li>x86_64</li><li>ARM 64</li></ul> |
     | <ul><li>Red Hat Enterprise Linux 7.3 or a later 7.x version</li><li>CentOS 7.3 or a later 7.x version</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>   |
+    | AWS Linux | <ul><li>x86_64</li><li>ARM 64</li></ul> |
```
Comment (dveeden) [inline hardware-and-software-requirements.md:24]:
```suggestion
    | Amazon Linux 2 | <ul><li>x86_64</li><li>ARM 64</li></ul> |
```

File: hardware-and-software-requirements.md [hunk 2]
```diff
@@ -19,7 +19,9 @@ Starting from v6.1.1, TiDB provides multi-level support for different quality st
     |  Operating systems   |   Supported CPU architectures   |
     |   :---   |   :---   |
     | <ul><li>Red Hat Enterprise Linux 8.4 or a later 8.x version</li><li>CentOS 8.4 or a later 8.x version</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
+    | Red Hat Enterprise Linux 8 Stream | <ul><li>x86_64</li><li>ARM 64</li></ul> |
```
Comment (dveeden) [inline hardware-and-software-requirements.md:22]:
```suggestion
    | CentOS 8 Stream | <ul><li>x86_64</li><li>ARM 64</li></ul> |
```

File: hardware-and-software-requirements.md [hunk 3]
```diff
@@ -19,7 +19,9 @@ Starting from v6.1.1, TiDB provides multi-level support for different quality st
     |  Operating systems   |   Supported CPU architectures   |
     |   :---   |   :---   |
     | <ul><li>Red Hat Enterprise Linux 8.4 or a later 8.x version</li><li>CentOS 8.4 or a later 8.x version</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
```
Comment (dveeden) [inline hardware-and-software-requirements.md:21]:
```suggestion
    | Red Hat Enterprise Linux 8.4 or a later 8.x version  |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
```

File: hardware-and-software-requirements.md [hunk 4]
```diff
@@ -19,7 +19,9 @@ Starting from v6.1.1, TiDB provides multi-level support for different quality st
...
+    | Red Hat Enterprise Linux 8 Stream | <ul><li>x86_64</li><li>ARM 64</li></ul> |
     | <ul><li>Red Hat Enterprise Linux 7.3 or a later 7.x version</li><li>CentOS 7.3 or a later 7.x version</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>   |
```
Comment (dveeden) [inline hardware-and-software-requirements.md:23]: Note that this line is still correct. RHEL7 and CentOS7 are both still supported.

**Comment by JasonWu0506** [comment]
LGTM

**Review by dveeden — APPROVED** [review]

**Pull request merged**